src/os/im/action/importaction.js

```diff
@@ -55,3 +55,20 @@ os.im.action.sortByLabel = function(a, b) {
   var bLabel = b ? b.getLabel() : '';
   return goog.array.defaultCompare(aLabel, bLabel);
 };
+
+
+/**
+ * Get columns from a vector layer.
+ *
+ * @param {os.layer.Vector} layer
+ * @return {Array} columns of the vector layer
+ */
+os.im.action.getVectorLayerColumns = function(layer) {
```

Comment:
I would suggest extracting this block and adapting it for use in this new function.
https://github.com/ngageoint/opensphere/blob/master/src/os/ui/im/action/filteractionimport.js#L121-L129

Your function definition will change a bit, as the input param should be a `os.filter.IFilterable` and it should return `Array<os.ogc.FeatureTypeColumn>`. I would also suggest renaming to reflect the new behavior, possibly `os.im.action.getColumnsFromFilterable`.

Once that's done, the function can be used in both `os.ui.im.action.FilterActionImportCtrl#onLayerChange` and `os.ui.im.action.FilterActionImporter#getFilterColumnsFromFilterable`.